### PR TITLE
feat(replays): Auto-dismiss the Replay>Network `req resp bodies are available` after the user sets them up

### LINKED ISSUE #48576: [Replay Details] Network Details panel 'new feature' alert should be forever hidden once replay bodies are setup
This is the alert:

Image

Currently it's only ever hidden once a user clicks the 'X'

In addition: we should hide it once users open the details panel and once they see a request or response payload (not headers tab)

### PR DESCRIPTION
**Pull request by ryan953**
**First** we show the promo message:
SCR-20230504-ngxr

**After there is data** we show the how-to-use messagae:
SCR-20230504-ngyx

And once they click into the request or response tab (when the tab has data, not just for any row in the list) then we will dismiss the alert so it won't appear next time they load a page.

Auto-dismissing the alert will not cause a re-render! So the alert stays up after they did the correct click. 

Fixes https://github.com/getsentry/sentry/issues/48576

### CODE REVIEW

File: static/app/views/replays/detail/network/details/onboarding.tsx [hunk 1]
```diff
@@ -1,16 +1,88 @@
...
+
+  const message = isNetworkDetailsSetup
+    ? tct(
+        'Click on a [fetch] or [xhr] request to see request and response payloads. [link]',
+        {
+          fetch: <code>fetch</code>,
+          xhr: <code>xhr</code>,
+          link: (
+            <ExternalLink
+              href="https://github.com/getsentry/sentry-javascript/issues/7103"
```
Comment (billyvg) [inline static/app/views/replays/detail/network/details/onboarding.tsx:43]: Hmm would this be the right link?

```suggestion
              href="https://docs.sentry.io/platforms/javascript/session-replay/configuration/#network-details"
```

File: static/app/views/replays/detail/network/details/onboarding.tsx [hunk 2]
```diff
@@ -1,16 +1,88 @@
...
+            >
+              {t('Learn More')}
+            </ExternalLink>
+          ),
+        }
+      )
+    : tct('Start collecting the body of requests and responses. [link]', {
+        link: (
+          <ExternalLink
+            href="https://github.com/getsentry/sentry-javascript/issues/7103"
```
Comment (billyvg) [inline static/app/views/replays/detail/network/details/onboarding.tsx:54]:
```suggestion
            href="https://docs.sentry.io/platforms/javascript/session-replay/configuration/#network-details"
```

File: static/app/views/replays/detail/network/details/sections.tsx [hunk 3]
```diff
@@ -101,7 +102,15 @@ export function QueryParamsSection({item}: SectionProps) {
 }
 
 export function RequestPayloadSection({item}: SectionProps) {
+  const {dismiss} = useDismissReqRespBodiesAlert();
+
   const hasRequest = 'request' in item.data;
+  useEffect(() => {
+    if (hasRequest) {
+      dismiss();
```
Comment (billyvg) [inline static/app/views/replays/detail/network/details/sections.tsx:110]: Is this a no-op if it's already dismissed?

**Review by billyvg — APPROVED** [review]

File: static/app/views/replays/detail/network/details/sections.tsx (hunk 3 above)
Comment (ryan953) [reply static/app/views/replays/detail/network/details/sections.tsx:110]: turns out no. it'll bump the timestamp each time. 👎

**Pull request merged**